# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/Worker.cpp
# Commit: 7802e5f1bd13
# Full Hash: 7802e5f1bd1395f19f2d78636787c41f64192bd7
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2024-04-19 15:22:44
# Regressor Bug: 1769913
# File Overlap Count: 3
# Description:
#   Bug 1769913 - P3 Remove WorkerRunnable::mWorkerPrivate. r=dom-worker-reviewers,asuth
#   
#   WorkerRunnable no longer keeps a raw pointer(mWorkerPrivate) for the associated WorkerPrivate in this patch.
#   Removing the WorkerRunnable::mWorkerPrivate needs to fix the following problems.
#   
# ==============================================================================

diff -r ffc69aa8f2c3 -r 7802e5f1bd13 dom/workers/Worker.cpp
--- a/dom/workers/Worker.cpp	Fri Apr 19 09:41:57 2024 +0000
+++ b/dom/workers/Worker.cpp	Fri Apr 19 09:41:58 2024 +0000
@@ -157,7 +157,7 @@
   // The worker could have closed between the time we entered this function and
   // checked ParentStatusProtected and now, which could cause the dispatch to
   // fail.
-  Unused << NS_WARN_IF(!runnable->Dispatch());
+  Unused << NS_WARN_IF(!runnable->Dispatch(mWorkerPrivate));
 }
 
 void Worker::PostMessage(JSContext* aCx, JS::Handle<JS::Value> aMessage,
@@ -191,7 +191,7 @@
     return;
   }
 
-  Unused << NS_WARN_IF(!aRunnable->Dispatch());
+  Unused << NS_WARN_IF(!aRunnable->Dispatch(mWorkerPrivate));
 }
 
 void Worker::Terminate() {